src/scripts/main.js

```diff
@@ -1,3 +1,223 @@
 'use strict';
 
-// write code here
+const table = document.querySelector('table');
+const body = table.querySelector('tbody');
+const rows = [...body.rows];
+const firstRow = rows[0];
+let currentEditingCell = null;
+
+function addForm() {
+  const form = document.createElement('form');
+  const button = document.createElement('button');
+  const selected = document.createElement('select');
+  const labelSelect = document.createElement('label');
+
+  const options = [
+    new Option('Tokyo', 'Tokyo'),
+    new Option('Singapore', 'Singapore'),
+    new Option('London', 'London'),
+    new Option('New York', 'New York'),
+    new Option('Edinburgh', 'Edinburgh'),
+    new Option('San Francisco', 'San Francisco'),
+  ];
+
+  options.forEach((option) => selected.add(option));
+
+  labelSelect.textContent = 'Office:';
+  labelSelect.append(selected);
+
+  form.className = 'new-employee-form';
+  button.textContent = 'Save to table';
+
+  const nameForm = addInput('name', 'text', 'Name', 'name');
+  const positionForm = addInput('position', 'text', 'Position', 'position');
+  const ageForm = addInput('age', 'number', 'Age', 'age');
+  const salaryForm = addInput('salary', 'number', 'Salary', 'salary');
+
+  form.append(nameForm, positionForm, labelSelect, ageForm, salaryForm, button);
+
+  button.addEventListener('click', (e) => {
+    e.preventDefault();
+
+    const nameEmploye = nameForm.querySelector('input').value.trim();
+    const position = positionForm.querySelector('input').value.trim();
+    const age = Number(ageForm.querySelector('input').value.trim());
+    const salary = Number(salaryForm.querySelector('input').value.trim());
+    const office = selected.value;
+
+    if (nameEmploye.length < 4) {
+      notification(false, 'Name Error', 'Name must be at least 4 characters.');
+
+      return;
+    }
+
+    if (age < 18 || age > 90) {
+      notification(false, 'Age Error', 'Age must be between 18 and 90.');
+
+      return;
+    }
+
+    addEmployeeToTable(nameEmploye, position, office, age, salary);
+    notification(true, 'Success', 'Employee added successfully.');
+    form.reset();
+  });
+
+  document.body.append(form);
+}
+
+function addEmployeeToTable(...args) {
+  const newRow = body.insertRow(-1);
+
+  args.forEach((value, index) => {
+    const newCell = newRow.insertCell(index);
+
+    if (index === 4) {
+      newCell.textContent = `$ ${value.toLocaleString()}`;
+    } else {
+      newCell.textContent = value;
+    }
+  });
+
+  rows.push(newRow);
+}
```

Comment:
The `addEmployeeToTable` function assumes that the number of arguments passed will always match the number of columns in the table. Consider adding a check to ensure that the number of arguments matches the expected number of columns to prevent potential errors.